packages/theme-check-common/src/checks/deprecated-global-app-block-type/index.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  LiquidBranchNamed,
+  LiquidConditionalExpression,
+  LiquidTag,
+  NamedTags,
+  NodeTypes,
+} from '@shopify/liquid-html-parser';
+import { Severity, SourceCodeType, LiquidCheckDefinition } from '../../types';
+
+export const DeprecatedGlobalAppBlockType: LiquidCheckDefinition = {
+  meta: {
+    name: 'DeprecatedGlobalAppBlockType',
+    code: 'deprecated-global-app-block-type',
+    severity: Severity.ERROR,
+    type: SourceCodeType.LiquidHtml,
+    docs: {
+      description: 'Check for deprecated global app block type `@global`',
+      url: 'https://shopify.dev/docs/themes/tools/theme-check/checks/deprecated-global-app-block-type',
+    },
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    function report(position: { start: number; end: number }) {
+      context.report({
+        message: 'The global app block type `@global` is deprecated. Use `@app` instead.',
+        startIndex: position.start,
+        endIndex: position.end,
+      });
+    }
+    function checkMarkup(markup: string | LiquidConditionalExpression): boolean {
+      if (typeof markup === 'string') {
+        return markup.includes('@global');
+      } else if (markup.type === 'Comparison') {
+        return checkMarkup(markup.left) || checkMarkup(markup.right);
+      } else if (markup.type === 'String') {
```

Comment:
In my understanding, the `NodeTypes.String` type is not applicable in this context because the markup property of the `LiquidBranch` node can be either a `string` or a` LiquidConditionalExpression` object. The` NodeTypes.String` type would be applicable if markup was always an object with a specific structure, but that's not the case here. Instead since we are checking the type of a variable we can use the typeof operator.

Reply:
Will add this change.